src/java.base/share/classes/sun/net/ftp/impl/FtpClient.java

```diff
@@ -1778,20 +1769,25 @@ public Date getLastModified(String path) throws sun.net.ftp.FtpProtocolException
         issueCommandCheck("MDTM " + path);
         if (lastReplyCode == FtpReplyCode.FILE_STATUS) {
             String s = getResponseString().substring(4);
-            Date d = null;
-            for (SimpleDateFormat dateFormat : dateFormats) {
-                try {
-                    d = dateFormat.parse(s);
-                } catch (ParseException ex) {
-                }
-                if (d != null) {
-                    return d;
-                }
-            }
+            return parseRfc3659TimeValue(s);
         }
         return null;
     }
 
+    private static Date parseRfc3659TimeValue(String s) {
```

Comment:
Shouldn't this method be synchronized because `SimpleDateFormat` is not thread-safe, but instances are stored in the static field `dateFormats`?
(Note though that this issue existed before as well)

Reply:
thanks @Marcono1234, it indeed needs to be `synchronized`, I'll take care of fixing that.

Reply:
@iignatev if you decide to log a followup a better alternative might be to use `[DateTimeFormatter](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/time/format/DateTimeFormatter.html)`